s3-vfs/src/test/java/org/pentaho/s3n/vfs/S3NFileObjectTest.java

```diff
@@ -248,7 +248,7 @@ public void testListChildrenRoot() throws FileSystemException {
   @Test
   public void testHandleAttachException() throws FileSystemException {
     AmazonS3Exception exception = new AmazonS3Exception( "NoSuchKey" );
-
+    exception.setErrorCode( "NoSuchKey" );
```

Comment:
Does this cause the 404 to occur? not sure if it needed to get coverage on the new logic.

Reply:
Nop. this was missing from the test, the test passed before because no exception was thrown. Now i do a getErrorCode, since it wasn't set a null pointer ended up being thrown, unrelated with what is being tested here.
Added a new test for the 'not found' behavior.